File: ecmascript/rna-transcription/rna-transcription.js
const DNA_TO_RNA = {
  'G': 'C',
  'C': 'G',
  'T': 'A',
  'A': 'U'
};

class Transcriptor {
  toRna(strand) {
    return [...strand].map((x) => DNA_TO_RNA[x]).join('');
  }
}

export default Transcriptor;

File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription
  (:require [clojure.string :as str]))

(def dna-to-rna
  {\G \C
   \C \G
   \T \A
   \A \U})

(defn- convert [nuc]
  (let [rna (dna-to-rna nuc)]
    (if (nil? rna)
      (throw (AssertionError. "Invalid nucleotide"))
      rna)))

(defn to-rna [strand]
  (str/join "" (map convert strand)))
